docs/annotations/index.md

```diff
@@ -116,6 +162,34 @@ In the previous example, the generated `resolve` config of the `something` field
 
 ## Type & Args auto-guessing
 
+When no explicit `type` is defined on a `@Field` (`@Query`, or `@Mutation`), the bundle will try to guess it using the following methods:
+- **Doc Block** The guess will be made from `@var` or `@return` in Doc block
+- **Type hint** The guess will be made from type hint 
+- **Doctrine annotations** The guess will be made from doctrine annotations
+
+The system will try every method, in the above order and return a type as soon as one of the guessing method return one.
```

Comment:
```suggestion
If the `type` option is not defined explicitly on the `@Field`, `@Query` or `@Mutation`, the bundle will try to guess it from other DocBlock annotations or from the PHP type-hint, in the following order:

1. `@var` and `@return` annotations
2. type-hint 
3. Doctrine annotations

It will stop on the first successful guess.
```

Just tried to simplify this. Maybe need some examples.